CMakeLists.txt

```diff
@@ -37,6 +37,12 @@ set(SOURCES
   src/Iterator.cpp
   src/WrappedIterator.cpp
   src/LimitIterator.cpp
+  src/ReverseIterator.cpp
+  src/RandomIterator.cpp
+  src/CountIterator.cpp
+  src/AndIterator.cpp
```

Comment:
```
AndIterator
OrIterator
DifferenceIterator
```
are keyed on `id()` instead of the `key()`. Its probably a good idea to document that.